lib/schema.js

```diff
@@ -213,10 +213,24 @@ Schema.prototype.refKeys = function(mdl) {
   for (var i = 0; i < refIndices.length; ++i) {
     var refIndex = refIndices[i];
     var refValues = [];
+    // This value will remain false unless a relevant value is found
+    var defined = false;
     for (var j = 0; j < refIndex.fields.length; ++j) {
-      refValues.push(this.fieldVal(mdl, refIndex.fields[j]));
+      var value = this.fieldVal(mdl, refIndex.fields[j]);
+      refValues.push(value);
+      //Prevents refdoc keys from having "undefined"
+      if(value){
```

Comment:
This should be `value !== undefined`, otherwise some kinds of valid input might be forgotten (0, false, "")